### PR TITLE
Test Driven Developer

### PR DESCRIPTION
**Pull request by coderacheal**
### **In this pull request**

- Write test before writing code
- Create a class Solver
- Create 3 methods,  factorial, reverse, fizzbuzz

### CODE REVIEW

File: specs/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    it 'should return 1 when 1 is passed' do
+      solver = Solver.new
+      expect(solver.factorial(1)).to eq(1)
+    end
+
+    it 'returns the correct value when given a positive integer' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eq(120)
+    end
+  end
```
Comment (BenMukebo) [inline specs/solver_spec.rb]: - [ ] Great job so far 👍🏻 however, The `#factorial` method should have tests that handle Negative value which raises an exception as an expectation. Kindly add this ✍🏻

File: specs/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      solver = Solver.new
+      expect(solver.fizzbuzz(9)).to eq('fizz')
+    end
+
+    it "should return 'buzz' if number is divisible by 3" do
+      solver = Solver.new
+      expect(solver.fizzbuzz(10)).to eq('buzz')
+    end
+
+    it "should return 'fizzbuzz' if number is divisible by 3" do
```
Comment (BenMukebo) [inline specs/solver_spec.rb]: - [ ] You should add 5 as well: "should return `'fizzbuzz' if number is divisible by 3 and 5"`

File: specs/solver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      solver = Solver.new
+      expect(solver.fizzbuzz(15)).to eq('fizzbuzz')
+    end
+  end
```
Comment (BenMukebo) [inline specs/solver_spec.rb:58]: - [ ] Same here, The `#fizzbuzz` method has tests that handle when a number N is not divisible by 3 or 5` which will expect to return that number. Kindly add this ✍🏻

File: README.md [hunk 4]
```diff
@@ -90,6 +90,12 @@ Install this project with:
 - GitHub: [@coderacheal](https://github.com/coderacheal)
 - LinkedIn: [racheal-appiah-kubi](https://www.linkedin.com/racheal-appiah-kubi/)
```
Comment (BenMukebo) [inline README.md]: - [ ] Dear. Please remember we are not expected to have any broken links in our readme file. Please fix the Ruby broken  link ⬇️ 

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @coderacheal, @hakichris,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- The tests cases written for `#factorial`, `#reverse` and `#fizzbuzz` are present✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯
- [x] Console app working as intended 🥇 
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**